optimum/intel/openvino/modeling_visual_language.py

```diff
@@ -780,16 +804,19 @@ def merge_vision_text_embeddings(
     def get_multimodal_embeddings(
         self, input_ids, pixel_values=None, attention_mask=None, position_ids=None, past_key_values=None, **kwargs
     ):
-        legacy_processing = (
-            not hasattr(self.config, "image_seq_length")
-            or ((input_ids == self.config.image_token_index).sum(1).max() < self.config.image_seq_length)
-            or (input_ids.shape[-1] == 1 and pixel_values is not None)
-        )
+        inputs_embeds = self.get_text_embeddings(input_ids, **kwargs)
+
+        if pixel_values is not None and self._support_new_processing and past_key_values is None:
+            legacy_processing = (input_ids == self.config.image_token_index).sum(
+                1
+            ).max() < self.config.image_seq_length
+            self._legacy_processing = legacy_processing
```

Comment:
Setting instance-level variable depending on a model input does not seem right. Can we somehow avoid it?

Reply:
Let me refresh how model behaves for the second token, if it keeps pixel_values even in generation cycle they are not used anymore, then yes, there is no need to use variable, if not  unfortunatly there is no way to avoid this